core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -185,6 +198,16 @@ List<ManifestFile> filterManifests(Schema tableSchema, List<ManifestFile> manife
       return ImmutableList.of();
     }
 
+    // Use the current set of referenced manifests as a source of truth when it's a subset of all
+    // manifests and all removals which were performed reference manifests.
+    // If a manifest is not in the trusted referenced set, filtering can be skipped
+    // assuming that the manifest does not have any live entries or aged out deletes
+    Set<String> manifestLocations =
+        manifests.stream().map(ManifestFile::path).collect(Collectors.toSet());
+    boolean trustReferencedManifests =
```

Comment:
Spotless formats this in a weird way. I wonder if we can play around with the names to stay on 1 line and/or potentially add a helper method.

Reply:
I tried a bit here on naming, but unfortunately couldn't get a shorter name which accurately captures what this variable represents. I just went ahead with a helper method, since I think really someone reading the code in most cases generally just needs to know what it means to trust the referenced manifests rather than the "how".  If they need to know the conditions for trust, it seems reasonable to just read the helper method logic.